internal/wasm/store.go

```diff
@@ -31,23 +31,17 @@ type (
 		ctx context.Context
 
 		// Engine is a global context for a Store which is in responsible for compilation and execution of Wasm modules.
-		Engine Engine
+		engine Engine
 
 		// EnabledFeatures are read-only to allow optimizations.
 		EnabledFeatures Features
 
 		// ModuleInstances holds the instantiated Wasm modules by module name from Instantiate.
-		ModuleInstances map[string]*ModuleInstance
-
-		// hostModules holds host functions by module name from NewHostModule.
-		hostModules map[string]*HostModule
-
-		// ModuleContexts holds default host function call contexts keyed by module name.
-		ModuleContexts map[string]*ModuleContext
+		moduleInstances map[string]*ModuleInstance
```

Comment:
I consolidate moduleContext and hostModule into a part of `ModuleInstance` to ease the management of each objects (under concurrent access)